client/src/modules/purchases/purchases.service.js

```diff
@@ -157,5 +158,37 @@ function PurchaseOrderService(
     return $httpParamSerializer(options);
   }
 
+  /**
+   * @method formatOptimalPurchase
+   *
+   * This Service analyzes for all existing repositories, Inventories that have reached a point of order,
+   * then calculates the overall quantity ordered for each Inventory
+   *
+   * @param {*} inventories - This is the list of all inventories
+   * @param {*} stock - it is the data of the inventories having reached their point of order
+   */
+  function formatOptimalPurchase(inventories, stock) {
+    inventories.forEach(inventory => {
+      inventory.unit_price = 0;
+      inventory._initialised = true;
+      inventory._hasValidAccounts = inventory.cogs_account;
```

Comment:
It looks like this is supposed to be a `Boolean` - https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/modules/purchases/create/PurchaseOrderItemService.js#L88

Reply:
I initialized this property just because when you place a purchase order, the property _hasValidAccounts contains the identifier of an account,  ![image](https://user-images.githubusercontent.com/8035588/68653973-38a18c80-052d-11ea-944e-de0b2f809361.png)

Reply:
Oh!  Than that is a bug.  It should return true if both a stock account and a cost of goods sold account exist.